internal/cli/server_upgrade.go

```diff
@@ -446,6 +448,38 @@ func (c *ServerUpgradeCommand) upgradeRunner(
 					runnerDefaultNames = append(runnerDefaultNames, fmt.Sprintf(runnerDefaultName, cfg.Name))
 					runnerUnsetStr = append(runnerUnsetStr, fmt.Sprintf(runnerUnsetDefault, cfg.PluginType, cfg.Name))
 				}
+
+				// Checking if the default runner profile for a platform, pre-0.9, has the correct task launcher configs
+				if c.platform == cfg.Name {
+					switch c.platform {
+					case "kubernetes":
+						// attempt to parse the runner profile config into the correct task launcher config struct
+						var result *k8s.TaskLauncherConfig
+						err = json.Unmarshal(cfg.PluginConfig, result)
+						if err != nil {
+							c.ui.Output(runnerProfileUpgradeConfigError, cfg.Name,
+								cfg.Name, clierrors.Humanize(err), terminal.WithWarningStyle())
+							var content map[string]interface{}
+							err = json.Unmarshal(cfg.PluginConfig, &content)
```

Comment:
I think so, this shouldn't happen, but it's possible the json config is malformed json, in which case this would error. So we should return an error for sure!